app/assets/javascripts/app/discussions/discussion_form.js

```diff
@@ -0,0 +1,46 @@
+(function() {
+  'use strict';
+
+  angular
+    .module('council.discussions')
+    .controller('DiscussionFormCtrl', DiscussionFormCtrl);
+
+  function DiscussionFormCtrl(Discussion, $state, $mdToast, $scope, discussion) {
+    var ctrl = this;
+
+    ctrl.nextScreen = nextScreen;
+    ctrl.discussion = discussion;
+    ctrl.createDiscussion = createDiscussion;
+    ctrl.updateDiscussion = updateDiscussion;
+
+    $state.go($state.current.name + '.step1');
+
+    function nextScreen(state) {
+      if ($scope.form.$valid)
+        $state.go(state);
+      else
+        showToast('Please fill the required fields');
+    }
+
+    function showToast(message) {
+      $mdToast.show(
+        $mdToast.simple()
+        .content(message)
+        .position('bottom right')
+        .hideDelay(2000)
+      );
+    }
+
+    function createDiscussion(discussion) {
+      Discussion.create(discussion).then(onDiscussionSave);
+    }
+
+    function updateDiscussion(discussion) {
+      Discussion.create(discussion).then(onDiscussionSave);
+    }
```

Comment:
Do we really need to the methods? Can't we rename it to something like `submitDiscussion` or `saveDiscussion`?